### PR TITLE
"Space lube" returns

### PR DESCRIPTION
**Pull request by EgorDinamit**
## About the Pull Request

- Re-adds space lube as a chemical available to emagged janitor cyborgs. It is also now named "Slippery motor oil".
- Changes overlay icons to be less invisible and represent just how badly lubed the floor is.
- Nerfed slip stun durations to make a bit more sense.

## Why It's Good For The Game

- Space lube is a funny non-lethal item that could stun you if you aren't paying attention, however, its previous iteration was rather overpowered (10 stun duration, bruh) and as such, I've readded it without all this bullshit.
- The overlays don't animate anymore, making it arguably a bit easier to spot a wet floor.
- Self-explanatory. Stun durations make more sense now.

## Did you test it?

Yes, it's funny and doesn't cause me to sit in stun for 10 years.

## Changelog

:cl:
rscadd: Slippery motor oil is now available to emagged janitor cyborgs.
tweak: Water/Lube stun durations are heavily nerfed so you won't spend 10 seconds lying on the floor.
tweak: The wet floor overlays don't become invisible every once in a while now, making it easier to spot a wet/lubed floor.
/:cl:

### CODE REVIEW

**Comment by Samuuu3** [comment]
I'm not a big fan of space lube, it seems like lrp garbage to me, and its very annoying to play around. (ping me on discord for more discussion)

**Comment by Cupax3** [comment]
Hello, as talked about on Discord, please adjust the name and description of the substance to be not space lube, thanks.

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Other.dm [hunk 1]
```diff
@@ -322,6 +322,23 @@
 	for (var/obj/effect/decal/cleanable/blood/B in T)
 		qdel(B)
 
+/datum/reagent/lube
+	name = "Slippery Motor Oil"
```
Comment (Shadowtail117) [inline code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Other.dm:326]:
```suggestion
	name = "slippery motor oil"
```

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Other.dm [hunk 2]
```diff
@@ -322,6 +322,23 @@
...
+	description = "An extra slippery motor oil. Meant to be used to reduce friction in the machinery components."
```
Comment (Shadowtail117) [inline code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Other.dm]: Condensity
```suggestion
	description = "An extra slippery motor oil. Designed to reduce friction in moving machinery parts."
```

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Other.dm [hunk 3]
```diff
@@ -322,6 +322,23 @@
...
+	taste_description = "slime"
```
Comment (Shadowtail117) [inline code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Other.dm]:
```suggestion
	taste_description = "motor oil"
```

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Other.dm [hunk 4]
```diff
@@ -322,6 +322,23 @@
 	for (var/obj/effect/decal/cleanable/blood/B in T)
 		qdel(B)
 
+/datum/reagent/lube
```
Comment (Shadowtail117) [inline code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Other.dm]: Please rename to something more fitting.

File: code/game/turfs/simulated.dm [hunk 5]
```diff
@@ -16,21 +16,21 @@
...
 		wet = wet_val
-		wet_overlay = image('icons/effects/water.dmi',src,"wet_floor")
+		wet_overlay = image('icons/effects/water.dmi',src,overlay_state)
 		overlays += wet_overlay
 
-	timer_id = addtimer(CALLBACK(src,/turf/simulated/proc/unwet_floor),8 SECONDS, TIMER_STOPPABLE|TIMER_UNIQUE|TIMER_NO_HASH_WAIT|TIMER_OVERRIDE)
+	timer_id = addtimer(CALLBACK(src,/turf/simulated/proc/unwet_floor, TRUE, dissipate_time), dissipate_time, TIMER_STOPPABLE|TIMER_UNIQUE|TIMER_NO_HASH_WAIT|TIMER_OVERRIDE)
 
-/turf/simulated/proc/unwet_floor(var/check_very_wet = TRUE)
+/turf/simulated/proc/unwet_floor(var/check_very_wet = TRUE, var/dissipate_timer = 8 SECONDS)
```
Comment (Shadowtail117) [inline code/game/turfs/simulated.dm]:
```suggestion
/turf/simulated/proc/unwet_floor(check_very_wet = TRUE, dissipate_timer = 8 SECONDS)
```

File: code/game/turfs/simulated.dm [hunk 6]
```diff
@@ -16,21 +16,21 @@
 	var/timer_id
 
 // This is not great.
-/turf/simulated/proc/wet_floor(var/wet_val = 1, var/overwrite = FALSE)
+/turf/simulated/proc/wet_floor(var/wet_val = 1, var/overwrite = FALSE, var/overlay_state = "wet_floor", var/dissipate_time = 8 SECONDS)
```
Comment (Shadowtail117) [inline code/game/turfs/simulated.dm]:
```suggestion
/turf/simulated/proc/wet_floor(wet_val = 1, overwrite = FALSE, overlay_state = "wet_floor", dissipate_time = 8 SECONDS)
```

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Other.dm [hunk 7]
```diff
@@ -322,6 +322,23 @@
...
+	name = "Slippery Motor Oil"
+	description = "An extra slippery motor oil. Meant to be used to reduce friction in the machinery components."
+	taste_description = "slime"
+	reagent_state = LIQUID
+	color = "#009ca8"
+	value = 0.6
+	should_admin_log = TRUE
+	var/overlay_icon = "lubed_floor"
+	var/lube_strength = 30
```
Comment (Shadowtail117) [inline code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Other.dm]:
```suggestion
	var/slip_strength = 30
```

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Other.dm [hunk 8]
```diff
@@ -322,6 +322,23 @@
...
+
+/datum/reagent/lube/touch_turf(var/turf/simulated/T)
```
Comment (Shadowtail117) [inline code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Other.dm]:
```suggestion
/datum/reagent/lube/touch_turf(turf/simulated/T)
```

**Review by Shadowtail117 — CHANGES_REQUESTED** [review]
As discussed, please ensure that people who are walking have a 0% chance to slip to address balance concerns.

Also, is the overlay the same as regular lube?

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Other.dm (hunk 1 above)
Comment (EgorDinamit) [reply code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Other.dm:326]: All reagent names are written in upper case.

**Comment by EgorDinamit** [comment]
PR updated. All suggestions from **resolved** conversations are in.

**Comment by EgorDinamit** [comment]
> As discussed, please ensure that people who are walking have a 0% chance to slip to address balance concerns.

The chance for slipping when walking is already as low as possible. If you wish to avoid slipping entirely - use creeping movement or rest.

> Also, is the overlay the same as regular lube?

Yes.

**Review by Shadowtail117 — CHANGES_REQUESTED** [review]
As echoed by Samuu, I am still iffy on the change even with the rename since even as an antagonist, cyborg-only thing its still a rather shitty antagonist tactic.

**Pull request merged**